nixos/modules/services/misc/weechat.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.weechat;
+in
+
+{
+  options.services.weechat = {
+    enable = mkEnableOption "weechat";
+    init = mkOption {
+      description = "Weechat commands applied at start, one command per line.";
+      example = ''
+        /set relay.network.password correct-horse-battery-staple
+        /relay add weechat 9001
+      '';
+      type = types.str;
+      default = "";
+    };
+    root = mkOption {
+      description = "Weechat state directory.";
+      type = types.str;
+      default = "/var/lib/weechat";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users = {
+      groups.weechat.gid = config.ids.gids.weechat;
+      users.weechat = {
+        createHome = true;
+        group = "weechat";
+	home = cfg.root;
+	uid = config.ids.uids.weechat;
+      };
+    };
+
+    systemd.services.weechat = {
+      environment.WEECHAT_HOME = cfg.root;
+      serviceConfig = {
+        User = "weechat";
+        Group = "weechat";
+      };
+      script = "exec ${pkgs.screen}/bin/screen -D -m ${pkgs.weechat}/bin/weechat";
```

Comment:
The use of `screen` is probably opinionated as is the usage of `tmux`. I personally prefer `tmux`.

Do you think adding an option (defaulted to `screen`) would be sensible?

Reply:
`tmux` doesn't work. Also, I don't think that this workaround warrants an option.

Reply:
I think screen is fine for now given the use case for this module. Not sure if you can attach to a running screen or tmux session without some hackery anyway. ;)

Reply:
Ideally, Weechat should be patched so that this hack is not required. It would be enough to strip ncurses initialization and run weechat without printing anything to terminal.

Reply:
Weechat 2.1 has a headless mode.